test/pylib/random_tables.py

```diff
@@ -256,16 +256,18 @@ def __str__(self):
 class RandomTables():
     """A list of managed random tables"""
     def __init__(self, test_name: str, manager: ManagerClient, keyspace: str,
-                 replication_factor: int):
+                 replication_factor: int, dc_replication_factor: dict[str, int]=None):
         self.test_name = test_name
         self.manager = manager
         self.keyspace = keyspace
         self.tables: List[RandomTable] = []
         self.removed_tables: List[RandomTable] = []
         assert self.manager.cql is not None
-        self.manager.cql.execute(f"CREATE KEYSPACE {keyspace} WITH REPLICATION = "
-                                 "{ 'class' : 'NetworkTopologyStrategy', "
-                                 f"'replication_factor' : {replication_factor} }}")
+        self.keyspace_query = f"CREATE KEYSPACE {keyspace} WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : {replication_factor}}}"
```

Comment:
Why do you want to put this variable in "self"?  Do you plan to use it anywhere except in this function?

Reply:
The idea here just not to repeat the same query twice. It used later in modifying the the replication factor for each DC in lines 267-270

Reply:
Yes, but all of these are **inside** the function, so this could have been a local variable, not an attribute of "self". Unless I'm missing something...

Reply:
By the way, @kostja maybe you want to review these changes to test/pylib so it doesn't "rot" in a way you don't like?

Reply:
I partly wrote them so I should be disqualified here. But I am OK if @xtrey submits the code that might "rot" a bit - he will bear the responsibility for the outcome, since I hope going forward he will be handling all test.py issues.
Thanks for pinging though.

Reply:
@nyh sorry, misunderstood you. You are completely right about moving it to the local variable.